### DISCUSSION TITLE
Multiple issues when trying to train models with my own dataset

### DISCUSSION THREAD

**Discussion by ep0p** [mindee/doctr#1662] [Q&A]
I have prepared my dataset following the required format however when training i encounter issue after issue:

1. The first one is this: [Bug](https://github.com/mindee/doctr/issues/1661) 
2. The second one, encountered while i bypassed the first one is now:

```  File "/home/epop/dataset/doctr/doctr/transforms/modules/pytorch.py", line 87, in forward
    target[:, [0, 2]] = offset[0] + target[:, [0, 2]] * raw_shape[-1] / img.shape[-1]
UnboundLocalError: local variable 'offset' referenced before assignment
```
These appear to be basic errors, such as variables not being initialized. I find it surprising that no one else has encountered these issues, considering that part of the code is two years old.

Am I doing something wrong?

**Comment by hanshupe007** [comment]
I have the same issue and obviously the code doesn't take care of the initialization. Did anyone find a solution for that?

**Reply by ep0p** [reply]
You will find the fix [here](https://github.com/mindee/doctr/discussions/1667)

https://github.com/mindee/doctr/blob/1cea7d8f7db2c3cf7365be3c7457b7fe67f82ab9/doctr/transforms/modules/pytorch.py#L80

```
 if target is not None:
     target = np.clip(target, 0, 1)
```
